jnagmp/src/main/java/com/squareup/jnagmp/LibGmp.java

```diff
@@ -53,20 +58,18 @@
   }
 
   private static void loadLibGmp() {
+    NativeLibrary library;
     try {
-      // Explicitly try to load the embedded version first.
-      File file = Native.extractFromResourcePath("gmp");
```

Comment:
+1.  I would suggest the following patch will do it...

```patch
diff --git a/jnagmp/src/main/java/com/squareup/jnagmp/LibGmp.java b/jnagmp/src/main/java/com/squareup/jnagmp/LibGmp.java
index 345f544..7d39800 100644
--- a/jnagmp/src/main/java/com/squareup/jnagmp/LibGmp.java
+++ b/jnagmp/src/main/java/com/squareup/jnagmp/LibGmp.java
@@ -55,7 +55,7 @@ public final class LibGmp {
   private static void loadLibGmp() {
     try {
       // Explicitly try to load the embedded version first.
-      File file = Native.extractFromResourcePath("gmp");
+      File file = Native.extractFromResourcePath("gmp", LibGmp.class.getClassLoader());
       load(file.getAbsolutePath());
       return;
     } catch (Exception ignored) {
@@ -66,7 +66,7 @@ public final class LibGmp {
   }
 
   private static void load(String name) {
-    NativeLibrary library = NativeLibrary.getInstance(name);
+    NativeLibrary library = NativeLibrary.getInstance(name, LibGmp.class.getClassLoader());
     Native.register(LibGmp.class, library);
     Native.register(SIZE_T_CLASS, library);
   }
```

I verified this change by running `mvn package` and then doing this:

```
~/github/jna-gmp java -cp jnagmp/target/jnagmp-1.0.1-SNAPSHOT.jar:$HOME/.m2/repository/net/java/dev/jna/jna/4.0.0/jna-4.0.0.jar:$HOME/.m2/repository/junit/junit/4.11/junit-4.11.jar:jnagmp/target/test-classes com.squareup.jnagmp.ModPowPerf
Testing 10000 512-bit private rsa's Java
10000 512-bit private rsa's in 2.132s
4690.43 ops/s
[...]
```